### DISCUSSION TITLE
Need a way to reset timeframe within the children of the timeline component

### DISCUSSION THREAD

**Discussion by colehart** [samuelarbibe/dnd-timeline#35] [Q&A] [answered]
Hello!

I am working on an implementation of the timeline that would enable a user to jump back to the default timeframe after scrolling to a point where the current time / time cursor is outside of the visible range / timeframe. It's a button that is on the TimeAxis so a child of the timeline > timeaxis component.

Is there an exposed function from useTimelineContext, useTimeline or useTimelineProps we could use to do that instead of trying to do some kind of deltaX / panEndEvent in a usePanStrategy?

**Comment by samuelarbibe** [comment, marked as answer]
Hey there!
The `timeframe` value is is fully controlled, and should not be modified by the library itself.  
It is passed in as an external value, and should be controlled and changed externally.  
Even the `usePanStrategy` does not modify the timeframe itself, but calls a provided callback. 
  
The simplest solution is to expose the `[timeframe, setTimeframe]` state and setter to the children that need it.  
For this purpose, you can use React's [Context API](https://react.dev/learn/passing-data-deeply-with-context), or my personal favorite: [jotai](https://jotai.org/)
